modules/platform/video/nuxeo-platform-video-rest/src/test/java/org/nuxeo/ecm/restapi/server/jaxrs/management/TestVideosObject.java

```diff
@@ -0,0 +1,227 @@
+        if (query != null) {
+            formData.add("query", query);
+        }
+        if (expectedRenditions != null) {
+            formData.put("conversionNames", expectedRenditions);
+        }
+
+        String commandId;
+        try (CloseableClientResponse response = httpClientRule.post("/management/videos/recompute/", formData)) {
+            if (missingConversionError) {
+                assertEquals(SC_BAD_REQUEST, response.getStatus());
+                return;
+            }
+            assertEquals(SC_OK, response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            assertBulkStatusScheduled(node);
+            commandId = getBulkCommandId(node);
+        }
+
+        // waiting for the asynchronous video renditions recompute task
+        txFeature.nextTransaction();
+        assertResponse(commandId, expectedRenditions, success);
+
+    }
+
+    protected void assertResponse(String commandId, List<String> expectedRenditions, boolean success)
+            throws IOException {
+        try (CloseableClientResponse response = httpClientRule.get("/management/bulk/" + commandId)) {
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            assertEquals(SC_OK, response.getStatus());
+            assertBulkStatusCompleted(node);
+            DocumentModel doc = session.getDocument(docRef);
+
+            var transcodedVideos = (List<Map<String, Serializable>>) doc.getPropertyValue(TRANSCODED_VIDEOS_PROPERTY);
+            assertNotNull(transcodedVideos);
+            if (success) {
+                assertEquals(1, node.get(STATUS_PROCESSED).asInt());
+                assertFalse(node.get(STATUS_HAS_ERROR).asBoolean());
+                assertEquals(0, node.get(STATUS_ERROR_COUNT).asInt());
+                assertEquals(1, node.get(STATUS_TOTAL).asInt());
+                assertTranscodedVideos(expectedRenditions, transcodedVideos);
+            } else {
+                assertEquals(0, node.get(STATUS_PROCESSED).asInt());
+                assertTrue(node.get(STATUS_HAS_ERROR).asBoolean());
+                assertEquals(1, node.get(STATUS_ERROR_COUNT).asInt());
+                assertEquals(0, node.get(STATUS_TOTAL).asInt());
+                assertEquals("Invalid query", node.get(STATUS_ERROR_MESSAGE).asText());
+                assertTrue(transcodedVideos.isEmpty());
+            }
+        }
+    }
+
+    protected void assertTranscodedVideos(List<String> expectedRenditions,
+            List<Map<String, Serializable>> transcodedVideos) {
+        if (expectedRenditions == null) {
+            VideoService videoService = Framework.getService(VideoService.class);
+            expectedRenditions = new ArrayList<>();
+            for (VideoConversion conversion : videoService.getAvailableVideoConversions()) {
+                expectedRenditions.add(conversion.getName());
+            }
```

Comment:
expectedRenditions = videoService.getAvailableVideoConversions()
                                     .stream()
                                     .map(VideoConversion::getName)
                                     .collect(Collectors.toList())

Which is the way you actually wrote it in `RecomputeRenditionsComputation` btw 😀

And personally I like to do a static import of `Collectors.toList` but it's up to you.